python/pipeline/pupil.py

```diff
@@ -915,42 +879,185 @@ def make(self, key):
+        """
+
+    @property
+    def key_source(self):
+        return (Tracking.proj() & 'tracking_method=2') * filter_config.OnlineMedianFilter.proj()
+
+    def make(self, key):
+        
+        print("Only for DLC tracked cases!")
+        print("Fitting:", key)
+
+        avi_path = (Eye & key).get_video_path()
+        nframes = (Eye & key).fetch1('total_frames')
+
+        data_circle = []
+        data_ellipse = []
+
+        dlc_config = (ConfigDeeplabcut & (
+            Tracking.Deeplabcut & key)).fetch1()
+
+        config = auxiliaryfunctions.read_config(dlc_config['config_path'])
+        config['config_path'] = dlc_config['config_path']
+        config['shuffle'] = dlc_config['shuffle']
+        config['trainingsetindex'] = dlc_config['trainingsetindex']
+
+        # find path to original video symlink
+        base_path = os.path.splitext(avi_path)[0] + '_tracking'
+        video_path = os.path.join(base_path, os.path.basename(avi_path))
+
+        config['orig_video_path'] = video_path
+
+        # find croppoing coords
+        cropped_coords = (Tracking.Deeplabcut & key).fetch1(
+            'cropped_x0', 'cropped_x1', 'cropped_y0', 'cropped_y1')
+
+        config['cropped_coords'] = cropped_coords
+
+        filter_dict = (filter_config.OnlineMedianFilter & key).fetch1()
+
+        pupil_fit = DLC_tools.DeeplabcutPupilFitting(
+            config=config, bodyparts='all', cropped=True, filtering=filter_dict)
+
+        for frame_num in tqdm(range(nframes)):
+
+            fit_dict = pupil_fit.fitted_core(frame_num=frame_num)
+
+            # circle info
+            if fit_dict['circle_fit']['center'] is None:
+                center_x, center_y = None, None
+            else:
+                center_x, center_y = fit_dict['circle_fit']['center']
+            radius = fit_dict['circle_fit']['radius']
+            visible_portion = fit_dict['circle_visible']['visible_portion']
+
+            data_circle.append(
+                [center_x, center_y, radius, visible_portion])
+
+            # ellipse info
+            if fit_dict['ellipse_fit']['center'] is None:
+                center_x, center_y = None, None
+            else:
+                center_x, center_y =  fit_dict['ellipse_fit']['center']
+
+            major_radius = fit_dict['ellipse_fit']['major_radius']
+            minor_radius = fit_dict['ellipse_fit']['minor_radius']
+            rotation_angle = fit_dict['ellipse_fit']['rotation_angle']
+            visible_portion = fit_dict['ellipse_visible']['visible_portion']
+
+            data_ellipse.append([center_x, center_y,
+                                major_radius, minor_radius,
+                                rotation_angle, visible_portion])
+
+        data_circle = np.array(data_circle).astype("float")
+        data_ellipse = np.array(data_ellipse).astype("float")
+
+        # now filter out the outliers by 5.5 std away from mean
```

Comment:
5.5 seems pretty specific any reason it's not 5 or 6?

Reply:
I tested out several cases and 6 is removing too little where as 5 could result in removing too many. well too many is also subject but not definitely 6 sigma. Hence, 5.5 was happy middle